crates/sparrow-merge/src/gather.rs

```diff
@@ -38,6 +42,15 @@ impl Gatherer {
         self.active.peek().map(|min| min.index)
     }
 
+    /// Returns a boolean indicating whether the gatherer can produce a batch.
+    pub fn can_produce(&self) -> bool {
```

Comment:
Could shorten to:

`self.active.peek().is_some_and(|top| top.up_to_time > self.emitted_up_to_time)`